platform/mellanox/mlnx-platform-api/sonic_platform/psu.py

```diff
@@ -169,3 +187,99 @@ def get_power(self):
+                    caps = psu_led_cap.read()
+                    cap_list = caps.split()
+        except (ValueError, IOError):
+            status = 0
+        
+        return cap_list
+
+
+    def set_status_led(self, color):
+        """
+        Sets the state of the PSU status LED
+
+        Args:
+            color: A string representing the color with which to set the
+                   PSU status LED
+
+        Returns:
+            bool: True if status LED state is set successfully, False if not
+
+        Notes:
+            Only one led for all PSUs.
+        """
+        led_cap_list = self._get_led_capability()
+        if led_cap_list is None:
+            return False
+
+        status = False
+        try:
+            if color == self.STATUS_LED_COLOR_GREEN:
+                with open(os.path.join(LED_PATH, self.psu_green_led_path), 'w') as psu_led:
+                    psu_led.write(str(LED_ON))
+                    status = True
+            elif color == self.STATUS_LED_COLOR_RED:
+                # Some fan don't support red led but support orange led, in this case we set led to orange
+                if self.STATUS_LED_COLOR_RED in led_cap_list:
+                    led_path = os.path.join(LED_PATH, self.psu_red_led_path)
+                elif self.STATUS_LED_COLOR_ORANGE in led_cap_list:
+                    led_path = os.path.join(LED_PATH, self.psu_orange_led_path)
+                else:
+                    return False
+                with open(led_path, 'w') as psu_led:
+                    psu_led.write(str(LED_ON))
+                    status = True
+            elif color == self.STATUS_LED_COLOR_OFF:
+                if self.STATUS_LED_COLOR_GREEN in led_cap_list:
+                    with open(os.path.join(LED_PATH, self.psu_green_led_path), 'w') as psu_led:
+                        psu_led.write(str(LED_OFF))
+                if self.STATUS_LED_COLOR_RED in led_cap_list:
+                    with open(os.path.join(LED_PATH, self.psu_red_led_path), 'w') as psu_led:
+                        psu_led.write(str(LED_OFF))
+                if self.STATUS_LED_COLOR_ORANGE in led_cap_list:
+                    with open(os.path.join(LED_PATH, self.psu_orange_led_path), 'w') as psu_led:
+                        psu_led.write(str(LED_OFF))
+
+                status = True
+            else:
+                status = False
+        except (ValueError, IOError):
+            status = False
+
+        return status
+
+
+    def get_status_led(self):
+        """
+        Gets the state of the PSU status LED
+
+        Returns:
+            A string, one of the predefined STATUS_LED_COLOR_* strings above
+        """
+        led_cap_list = self._get_led_capability()
+        if led_cap_list is None:
+            return self.STATUS_LED_COLOR_OFF
+
+        try:
+            with open(os.path.join(LED_PATH, self.psu_green_led_path), 'r') as psu_led:
```

Comment:
is green assumes to be there always? why not using if statement as the others?

Reply:
yes. the green should be always supported.